src/Training/Bundle/UserNamingBundle/Twig/TrainingFormatExtension.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Training\Bundle\UserNamingBundle\Twig;
+
+use Oro\Bundle\UserBundle\Entity\User;
+use Training\Bundle\UserNamingBundle\Formatter\UserNameFormatter;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+class TrainingFormatExtension extends AbstractExtension
+{
+    private UserNameFormatter $nameFormatter;
+
+    public function __construct(UserNameFormatter $nameFormatter)
+    {
+        $this->nameFormatter = $nameFormatter;
+    }
+
+    public function getFilters()
+    {
+        return [
+            new TwigFilter('training_example_user_naming', [$this, 'getExampleForUserNaming']),
+        ];
+    }
+
+    public function getExampleForUserNaming(string $format): string
```

Comment:
Twig extensions are loaded on every twig call with all the dependencies, so it generally makes sense to make them lazy-loaded in the Oro application, as there are too many of them. 
https://symfony.com/doc/current/templating/twig_extension.html#creating-lazy-loaded-twig-extensions